### ISSUE TITLE
pull container images through the internal module image-rs

### ISSUE THREAD

**Issue by ccxiaop** [kata-containers/kata-containers#9745]
I am currently using kata 3.3 version. I want to be able to use k8s to pull container images through the internal module image-rs of kata agent. What do I need to do? Is there a corresponding guidance document?

**Comment by ccxiaop** [comment]
I found that in kata3.3, the image.rs file has the pull_image function, but I don't know how to call this function to pull the image in the guest. Can you tell me how to pull the image in the guest? Thank you, sir

**Comment by mkulke** [comment]
The image-pull feature is gated behind the `guest-pull` flag. The logic currently depends on the containerd remote-snapshotter feature and the project is being tested with nydus, which you need to set up in your cluster.

**Comment by Martinits** [comment]
@mkulke Hi, I'm trying run confidential containers with Kata 3.6.0 on TDX. Can I pull images inside guest without nydus? Using nydus may cause extra issues in my situation. Thanks.

**Comment by mkulke** [comment]
No, for time being a remote snapshotter is required for in-guest pulling containerd, I think it might be possible with cri-o? cc @littlejawa
